test/e2e/density.go

```diff
@@ -203,13 +199,21 @@ var _ = Describe("Density", func() {
 		// during the test so clean it up here
 		rc, err := c.ReplicationControllers(ns).Get(RCName)
 		if err == nil && rc.Spec.Replicas != 0 {
-			DeleteRC(c, ns, RCName)
+			By("Cleaning up the replication controller")
```

Comment:
This seems much better in the defer, which will ensure that it's cleaned up no matter what. If you want to not clean it up if it didn't get created, I think it's better to make an ```var rcCreated bool``` and conditioning the teardown on that.